singer/src/main/java/com/pinterest/singer/utils/SingerUtils.java

```diff
@@ -96,15 +96,15 @@ public static String getHostname() {
   }
 
   /***
-   * Gradually builds substrings from hostname separated by dashes
-   * will return hostname if hostname does not contain dashes
+   * Gradually builds substrings from hostname separated by a given regex,
+   * will return hostname if hostname can't be split by regex
    *
    * @param
    * @return a list of hostname prefixes
    */
-  public static List<String> getHostnamePrefixes() {
+  public static List<String> getHostnamePrefixes(String regex) {
```

Comment:
Can you please add unit tests for this function?

Reply:
Done